File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        maxArea = 0
        left = 0
        right = len(height)-1
        while(left < right):
            l = right - left
            ht = min(height[left], height[right])
            area = l*ht
            maxArea = max(maxArea, area)
            while(left < right and height[left] <= ht):
                left += 1
            while(left < right and height[right] <= ht):
                right -= 1

        return maxArea
        

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int maxArea = 0;
        int left = 0;
        int right = height.length-1;
        while(left < right){
            int len = right-left;
            int ht = Math.min(height[left], height[right]);
            int area = len*ht;
            maxArea = Math.max(area, maxArea);

            while(left < right && height[left] <= ht) left++;
            while(left < right && height[right] <= ht) right--;
        }
        return maxArea;
    }
}